packages/utils/src/schema/getDefaultFormState.ts

```diff
@@ -255,41 +263,53 @@ export function computeDefaults<T = any, S extends StrictRJSFSchema = RJSFSchema
   switch (getSchemaType<S>(schema)) {
     // We need to recurse for object schema inner default values.
     case 'object': {
-      const objectDefaults = Object.keys(schema.properties || {}).reduce((acc: GenericObjectType, key: string) => {
-        // Compute the defaults for this node, with the parent defaults we might
-        // have from a previous run: defaults[key].
-        const computedDefault = computeDefaults<T, S, F>(validator, get(schema, [PROPERTIES_KEY, key]), {
-          rootSchema,
-          _recurseList,
-          experimental_defaultFormStateBehavior,
-          includeUndefinedValues: includeUndefinedValues === true,
-          parentDefaults: get(defaults, [key]),
-          rawFormData: get(formData, [key]),
-          required: schema.required?.includes(key),
-        });
-        maybeAddDefaultToObject<T>(
-          acc,
-          key,
-          computedDefault,
-          includeUndefinedValues,
-          required,
-          schema.required,
-          experimental_defaultFormStateBehavior
-        );
-        return acc;
-      }, {}) as T;
-      if (schema.additionalProperties) {
+      // This is a custom addition that fixes this issue:
+      // https://github.com/rjsf-team/react-jsonschema-form/issues/3832
+      const retrievedSchema =
+        experimental_defaultFormStateBehavior?.allOf === 'populateDefaults' && ALL_OF_KEY in schema
+          ? retrieveSchema<T, S, F>(validator, schema, rootSchema, formData)
+          : schema;
+      const objectDefaults = Object.keys(retrievedSchema.properties || {}).reduce(
```

Comment:
I forget if it is possible for `allOf` to be in something other than the object type?

Reply:
@benjdlambert Do you know whether this is possible? I can imagine if it is, then this solution may be incomplete

Reply:
Hmm, so I guess you mean something like this right?

```ts
const schema = { anyOf: [{ type: 'string' }, { type: 'number' }] }
```

Or do you have an example schema in mind that I can test with.

Reply:
That's an `anyOf`. I'm guessing that you probably haven't seen an JSON Schema that has an `allOf` that isn't part of an `object` type?

Reply:
Anyway, this is better than what we had before, so I'm going to approve

Reply:
Yeah, actually i'm not sure. I've never seen one, not sure if there's any other datatypes where it would be possible to use `allOf` like that. Happy to provide a followup if there are!